vispy/visuals/text/text.py

```diff
@@ -183,27 +166,113 @@ def _text_to_vbo(text, font, anchor_x, anchor_y, lowres_size):
+                #width -= glyph['advance'] * ratio - (glyph['size'][0] - 2*slop)
+
+                if anchor_x == 'right':
+                    dx = -width
+                elif anchor_x == 'center':
+                    dx = -width / 2.
+                vertices['a_position'][vi_marker:vi+4] += (dx, dy)
+                vi_marker = vi+4
+                ii_offset -= 1
+                # Reset varialbles that affects x-direction positioning
+                x_off = -slop
+                width = 0
+                # Add offset in y-direction
+                y_offset += esc_seq[ord(char)] * lineheight
+        else:
+            # For ordinary characters, normal procedure
+            glyph = font[char]
+            kerning = glyph['kerning'].get(prev, 0.) * ratio
+            x0 = x_off + glyph['offset'][0] * ratio + kerning
+            #NB y0 = glyph['offset'][1] * ratio + slop
+            y0 = glyph['offset'][1] * ratio + slop - y_offset
+            x1 = x0 + glyph['size'][0]
+            y1 = y0 - glyph['size'][1]
+            u0, v0, u1, v1 = glyph['texcoords']
+            position = [[x0, y0], [x0, y1], [x1, y1], [x1, y0]]
+            texcoords = [[u0, v0], [u0, v1], [u1, v1], [u1, v0]]
+            #NB vi = ii * 4
+            vi = (ii + ii_offset) * 4
+            vertices['a_position'][vi:vi+4] = position
+            vertices['a_texcoord'][vi:vi+4] = texcoords
+            x_move = glyph['advance'] * ratio + kerning
+            x_off += x_move
+            ascender = max(ascender, y0 - slop)
+            descender = min(descender, y1 + slop)
+            width += x_move
+            height = max(height, glyph['size'][1] - 2*slop)
+            prev = char
+
     if orig_viewport is not None:
         canvas.context.set_viewport(*orig_viewport)
 
     # Tight bounding box (loose would be width, font.height /.asc / .desc)
-    width -= glyph['advance'] * ratio - (glyph['size'][0] - 2*slop)
+    #width -= glyph['advance'] * ratio - (glyph['size'][0] - 2*slop)
+
+    #NB I changed the dx and dy to what seems reasonable to me
     dx = dy = 0
     if anchor_y == 'top':
-        dy = -ascender
+        dy = -descender
     elif anchor_y in ('center', 'middle'):
-        dy = -(height / 2 + descender)
+        #dy = -(height / 2 + descender)
+        dy = (-descender - ascender) / 2
     elif anchor_y == 'bottom':
-        dy = -descender
+        dy = -ascender
+
     # Already referenced to baseline
     # elif anchor_y == 'baseline':
     #     dy = -descender
     if anchor_x == 'right':
         dx = -width
     elif anchor_x == 'center':
         dx = -width / 2.
-    vertices['a_position'] += (dx, dy)
+
+    #NB If any linebreaks occured in text, we only want to translate characters
+    #   in the last line in text (those after the vi_marker)
+    #vertices['a_position'] += (dx, dy)
+    vertices['a_position'][0:vi_marker] += (0, dy)
+    vertices['a_position'][vi_marker:] += (dx, dy)
```

Comment:
Does this work even if there are multiple line breaks in the text?

Reply:
I will edit the example in the pull request during the evening. Any other questions or things to discuss?

Reply:
@djhoese

Reply:
The only issue I see is that there are some minimal "air" around the text in respect to it's placement coordinate (if right or left align). But that is very minimal and very hard to see if using regular text size.

Reply:
Haha, I will do. NB (nota bene) is from Latin and regularly used by us Norwegians, I don't know about you foreigners ;)

Reply:
I have mostly seen it used in mathematics circles, I'm fine with it :)

Reply:
I will remove most of the "NB"s, I only used them because they got highlighted by the text editor I used before pycharm.